### PR TITLE
Updates DRA scripts to build snapshot artifacts

### PR DESCRIPTION
**Pull request by andsel**
## Release notes

[rn:skip] 

## What does this PR do?

Handle the `WORKFLOW_TYPE` enviroment variable used to select the kind of artifacts to generate and consequently adapt the version name.
If the `WORKFLOW_TYPE` has a value assigned other then empty string it's assumed to be `snapshot` and so it generates snapshot artifacts else the release ones.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files (and/or docker env variables)~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## Related issues

-

### CODE REVIEW

File: ci/dra_aarch64.sh [hunk 1]
```diff
@@ -9,26 +9,49 @@ export JRUBY_OPTS="-J-Xmx1g"
...
-  VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" RELEASE=1 rake artifact:docker
-  VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" RELEASE=1 rake artifact:docker_oss
-  VERSION_QUALIFIER="$VERSION_QUALIFIER_OPT" rake artifact:dockerfiles
-  # Qualifier is passed from CI as optional field and specify the version postfix
-  # in case of alpha or beta releases:
-  # e.g: 8.0.0-alpha1
-  STACK_VERSION="${STACK_VERSION}-${VERSION_QUALIFIER_OPT}"
+  # WORKFLOW_TYPE is set, it has value "SNAPSHOT"
+  if [ -z "$VERSION_QUALIFIER_OPT" ]; then
+    RELEASE=1 rake artifact:docker
```
Comment (kaisecheng) [inline ci/dra_aarch64.sh:37]: Do we need RELEASE=1 for snapshot?

File: ci/dra_aarch64.sh [hunk 2]
```diff
@@ -9,26 +9,49 @@ export JRUBY_OPTS="-J-Xmx1g"
...
-STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:\s\([[:digit:]]*\.[[:digit:]]*\.[[:digit:]]*\)$/\1/p'`
+STACK_VERSION=`cat versions.yml | sed -n 's/^logstash\:[[:space:]]\([[:digit:]]*\.[[:digit:]]*\.[[:digit:]]*\)$/\1/p'`
 
-if [ -z "$VERSION_QUALIFIER_OPT" ]; then
-  RELEASE=1 rake artifact:docker
-  RELEASE=1 rake artifact:docker_oss
-  rake artifact:dockerfiles
+# WORKFLOW_TYPE is a CI externally configured environment variable where, if not an empty string,
+# it's assumed to be snapshot
+if [ -z "$WORKFLOW_TYPE"]; then
```
Comment (kaisecheng) [inline ci/dra_aarch64.sh:16]: The comments said if not empty, it is a snaphot, but here it says `WORKFLOW_TYPE` is empty and append snapshot.
Do you want to check with `-n` for non empty?

Comment (andsel) [reply ci/dra_aarch64.sh:16]: You are correct! it should be `-n`

File: ci/dra_aarch64.sh (hunk 1 above)
Comment (andsel) [reply ci/dra_aarch64.sh:37]: Yes you are right, I fooled myself with the ifs :-(

File: ci/dra_upload.sh [hunk 3]
```diff
@@ -18,6 +18,12 @@ if [ -n "$VERSION_QUALIFIER_OPT" ]; then
   STACK_VERSION="${STACK_VERSION}-${VERSION_QUALIFIER_OPT}"
 fi
 
+WORKFLOW="staging"
+if [ -z "$WORKFLOW_TYPE"]; then
```
Comment (kaisecheng) [inline ci/dra_upload.sh:22]:
```suggestion
if [ -n "$WORKFLOW_TYPE"]; then
```

**Review by kaisecheng — COMMENTED** [review]
I left one comment, otherwise looks good to me

File: ci/dra_aarch64.sh (hunk 2 above)
Comment (andsel) [reply ci/dra_aarch64.sh]: I've fixed it

File: ci/dra_upload.sh (hunk 3 above)
Comment (andsel) [reply ci/dra_upload.sh:22]: Fixed

**Comment by andsel** [comment]
Jenkins test this please

File: ci/dra_upload.sh [hunk 4]
```diff
@@ -18,6 +18,12 @@ if [ -n "$VERSION_QUALIFIER_OPT" ]; then
   STACK_VERSION="${STACK_VERSION}-${VERSION_QUALIFIER_OPT}"
 fi
 
+WORKFLOW="staging"
+if [ -n "$WORKFLOW_TYPE"]; then
```
Comment (kaisecheng) [inline ci/dra_upload.sh:22]:
```suggestion
if [ -n "$WORKFLOW_TYPE" ]; then
```

**Review by kaisecheng — APPROVED** [review]
LGTM

**Comment by andsel** [comment]
@logstashmachine backport 8.5

**Comment by andsel** [comment]
@logstashmachine backport 7.17

**Pull request merged**